syntax = "proto3";
package transport_router_serialization;

message Edge {
  uint32 from = 1;
  uint32 to = 2;
  double weight = 3;
  int32 span_count = 4;
  bool is_bus = 5;
  string bus_name = 6;
}

message IncidenceList {
  repeated uint32 edge_id = 1;
}

message Graph {
  repeated IncidenceList incidence_lists = 1;
  repeated Edge edges = 2;
}